### PR TITLE
[Feature] Create new Evaluator base and `GSgnnClassificationEvaluator`

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #757 , if available:*

*Description of changes:*
This PR creates a new base class, `GSgnnBaseEvaluator`, which integrates the `GSgnnInstanceEvaluator` and `GSgnnLPEvaluator` as one.

Based on this new base class, creates a new `GSgnnClassificationEvalutor`, which set "accuracy" as the default evaluation metric, and False to `multilabel` argument.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/eval/evaluator.py [hunk 1]
```diff
@@ -108,9 +108,386 @@ def get_val_score_rank(val_score, val_perf_rank_list, comparator):
     return rank
 
 
+class GSgnnBaseEvaluator():
```
Comment (classicsong) [inline python/graphstorm/eval/evaluator.py:111]: This class is very close to GSgnnInstanceEvaluator, why not just rename GSgnnInstanceEvaluator and update it?

File: python/graphstorm/eval/evaluator.py [hunk 2]
```diff
@@ -108,9 +108,386 @@ def get_val_score_rank(val_score, val_perf_rank_list, comparator):
...
+            if test_labels is not None else None
+
+        with th.no_grad():
+            val_score = self.compute_score(val_pred, val_labels, train=False)
+            test_score = self.compute_score(test_pred, test_labels, train=False)
+
+        for metric in self.metric:
+            # be careful whether > or < it might change per metric.
+            if self.metrics_obj.metric_comparator[metric](
+                self._best_val_score[metric],val_score[metric]):
```
Comment (classicsong) [inline python/graphstorm/eval/evaluator.py:443]:
```suggestion
                self._best_val_score[metric], val_score[metric]):
```

File: python/graphstorm/eval/evaluator.py [hunk 3]
```diff
@@ -584,7 +964,7 @@ def compute_score(self, pred, labels, train=True):
         return results
 
 class GSgnnLPEvaluator():
-    """ Template class for user defined evaluator.
+    """ Base class for user defined Link Prediction evaluator.
```
Comment (classicsong) [inline python/graphstorm/eval/evaluator.py:967]: Shouldn't GSgnnLPEvaluator inherit from GSgnnBaseEvaluator?

File: python/graphstorm/eval/evaluator.py [hunk 4]
```diff
@@ -466,6 +843,9 @@ def compute_score(self, pred, labels):
 class GSgnnAccEvaluator(GSgnnInstanceEvaluator):
```
Comment (classicsong) [inline python/graphstorm/eval/evaluator.py:843]: Do you want to remove this function in this PR?

File: tests/unit-tests/test_evaluator.py [hunk 5]
```diff
@@ -375,6 +375,145 @@ def check_evaluate_infer(mock_compute_score):
 
     th.distributed.destroy_process_group()
 
+def test_classification_evaluator():
```
Comment (classicsong) [inline tests/unit-tests/test_evaluator.py:378]: Is this almost same as test_acc_evaluator?

File: python/graphstorm/eval/__init__.py [hunk 6]
```diff
@@ -26,5 +26,5 @@
 from .evaluator import GSgnnInstanceEvaluator
 from .evaluator import GSgnnLPEvaluator
 from .evaluator import GSgnnMrrLPEvaluator, GSgnnPerEtypeMrrLPEvaluator
-from .evaluator import GSgnnAccEvaluator
+from .evaluator import GSgnnAccEvaluator, GSgnnClassificationEvaluator
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply python/graphstorm/eval/__init__.py:29]: I will remove these "Deprecated" class in this refactor.

File: python/graphstorm/eval/evaluator.py (hunk 1 above)
Comment (zhjwy9343) [reply python/graphstorm/eval/evaluator.py:111]: I want to make sure the existing code will not be broke, and then replace the deprecated Evaluators one by one.

File: python/graphstorm/eval/evaluator.py (hunk 3 above)
Comment (zhjwy9343) [reply python/graphstorm/eval/evaluator.py:967]: Yes, will create a new GSgnnLinkPredictionEvaluator to replace the GSgnnLPEvaluator in the next PR.

File: python/graphstorm/eval/evaluator.py (hunk 4 above)
Comment (zhjwy9343) [reply python/graphstorm/eval/evaluator.py:843]: No. I would like to remove all deprecated tests after all Evaluators are replaced with new ones.

File: tests/unit-tests/test_evaluator.py (hunk 5 above)
Comment (zhjwy9343) [reply tests/unit-tests/test_evaluator.py:378]: Yes, it is very similar. Will remove it after all replacements implemented.

File: python/graphstorm/eval/evaluator.py (hunk 2 above)
Comment (zhjwy9343) [reply python/graphstorm/eval/evaluator.py:443]: changed in local branch.

**Comment by zhjwy9343** [comment]
Per discussion, will close this PR. And create a dev branch and PR there before finally merge dev branch to the main.

**Pull request closed without merging**